src/viam/app/data/client.py

```diff
@@ -41,37 +55,149 @@ def __init__(self, channel: Channel, metadata: str):
 
     async def tabular_data_by_filter(
         self,
-        filter: Optional[Filter],
-        dest: Optional[str]
+        filter: Optional[Filter] = None,
```

Comment:
If all these `filter` params are optional (here and in the other methods), I think I'd prefer renaming these to simply be `tabular_data` or `binary_data` (dropping the `by_filter`). Especially since filter is optional with a default to `None`. 

Thoughts?

Reply:
Ooh, good point. Hmm... I agree that `{tabular|binary}_data` is a better name in the abstract, but I think I have a weak preference for keeping the names as they are. Changing would create inconsistency with the other data getter methods, the `*_by_filter` data delete methods, and the proto code itself. 

That said, I don't feel particularly strongly about this so happy to go with the shorter names if you do!

Reply:
I think the difference is that `by_id` REQUIRES an ID (or list of IDs), whereas these do not have any required params. Also, I don't think it's necessary to exactly match the proto names -- I do think the names should be descriptive in the language they are in.

That being said, I also don't have a super strong preference. I just want to make sure we discuss this prior to pushing, because I don't want to make a breaking change once this is released.

Reply:
Yeah, totally valid. I think my main focus was on consistency with the `delete_by_filter` methods (where the filter is technically optional but I opted for not providing a default because I don't want to make it too easy for folks to delete everything). So people do have to provide _something_ for their filter (even if it is a `None`).

I very much agree that the names should be descriptive for their language, so if you think dropping the `by_filter` from the getters (and possibly the deleters?) is more Python idiomatic then I will readily defer to that judgment.

Reply:
Hmmm... the `delete_by_filter` really makes me think we should keep these `data_by_filter` as well... Ok this is fine, let's keep it for now and if we want to change it we can always alias it with a deprecation warning